### PR TITLE
Fixing issue with JarOrder optimisation while building on Windows

### PR DESCRIPTION
**Pull request by ikishore**
Due to different path notation of linux and windows, Modules/Jars mapping with module path was not correct and optmization was not working while building the intellij community version in Windows. 
Fixed the issue by converting paths to Windows format while building in Windows. Verified the change by building the version in Windows.

### CODE REVIEW

**Comment by anstarovoyt** [comment]
@ikishore Hi! Thank you for the PR. I am not sure that it is the best way to implement it. I think it is better to store in ``pathToToJarName``/``pathToModuleName`` normalized (system-independent) paths and keep all the usages of the collections "as is".

**Comment by ikishore** [comment]
I will 

> @ikishore Hi! Thank you for the PR. I am not sure that it is the best way to implement it. I think it is better to store in `pathToToJarName`/`pathToModuleName` normalized (system-independent) paths and keep all the usages of the collections "as is".

True, I will evaluate if those methods are not used any where else and modify the patch accordingly

**Comment by ikishore** [comment]
@anstarovoyt  Submitted the patch, Please review and let me know if any further changes required. Thanks

**Comment by ikishore** [comment]
Hi @anstarovoyt Please review the patch and let me know if any further changes required, thanks

**Comment by anstarovoyt** [comment]
@ikishore sorry, I'v added comments but haven't submitted the review 🤦

File: platform/build-scripts/groovy/org/jetbrains/intellij/build/impl/DistributionJARsBuilder.groovy [hunk 1]
```diff
@@ -547,7 +548,8 @@ class DistributionJARsBuilder {
             jarName = candidate
           }
         }
-        libraryJarPathToJarName.put(libFile.getPath(), jarName)
+        def jarPath = (isWindows()) ? '/' + libFile.getPath().replace('\\', '/') : libFile.getPath();
```
[the remark this answers is not in the document]
Comment (ikishore) [reply platform/build-scripts/groovy/org/jetbrains/intellij/build/impl/DistributionJARsBuilder.groovy:551]: Please confirm below change would be enough? I feel adding '/' prefix cannot be avoided still. Once you confirm the changes I will submit the patch, Thanks
def jarPath = (isWindows()) ? '/' + FileUtil.toSystemIndependentName(libFile.getPath()) : libFile.getPath();

[the remark this answers is not in the document]
Comment (anstarovoyt) [reply platform/build-scripts/groovy/org/jetbrains/intellij/build/impl/DistributionJARsBuilder.groovy:551]: yes, I think so

**Comment by anstarovoyt** [comment]
@ikishore also  have you submitted a contributor license agreement? If not, please follow the steps described at http://www.jetbrains.com/agreements/cla/ to sign it.

**Comment by ybedrov** [comment]
@anstarovoyt  CLA is signed

**Comment by ikishore** [comment]
@anstarovoyt submitted the changes, Please review. Thanks

**Comment by anstarovoyt** [comment]
@ikishore the last question: could you please squash the commits into a single one? And after that, I will be ready to merge it in master.

**Comment by ikishore** [comment]
ok sure

**Comment by ikishore** [comment]
lot of conflicts while rebasing

**Pull request closed without merging**